app/components/project/ProjectMap.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/consistent-type-imports */
+import { useEffect, useState, useRef } from "react";
+import { SlotURI } from "~/types/slotURI";
+import "maplibre-gl/dist/maplibre-gl.css";
+import { RMap, RMarker } from "maplibre-react-components";
+import maplibregl from "maplibre-gl";
+import "maplibre-react-components/style.css";
+import "maplibre-theme/modern.css";
+import "maplibre-theme/icons.default.css";
+import CustomMarker from "./CustomMarker";
+import MapSkeleton from "../common/MapSkeleton";
+
+interface ProjectMapProps {
+    mapData: SlotURI[];
+}
+
+function filterByName(arr: SlotURI[]) {
+    const uniqueNames = new Set();
+    return arr.filter(item => {
+        if (!uniqueNames.has(item.name)) {
+            uniqueNames.add(item.name);
+            return true;
+        }
+        return false;
+    });
+}
+
+const geocodeLocation = async (location: string) => {
+    const response = await fetch(
+        `https://nominatim.openstreetmap.org/search?q=${location}&format=json`
+    );
+    const data = await response.json();
+    if (data.length > 0) {
+        return [parseFloat(data[0].lon), parseFloat(data[0].lat)]; // [longitude, latitude]
+    }
+    return null;
+};
+
+export default function ProjectMap({ mapData }: ProjectMapProps) {
+    const mapRef = useRef<maplibregl.Map | null>(null);
+    useEffect(() => {
+        if (mapData) {
+            console.log(filterByName(mapData));
```

Comment:
Is there some code missing here instead of the console.log?

Reply:
No nothing was missing here I was testing the filter function